File: spring-web-java/src/main/java/spring/web/java/infrastructure/persistence/adapter/PermissionRepositoryAdapter.java
package spring.web.java.infrastructure.persistence.adapter;

import java.util.Optional;

import org.springframework.stereotype.Repository;

import lombok.RequiredArgsConstructor;
import spring.web.java.domain.model.entity.Permission;
import spring.web.java.domain.repository.PermissionRepository;
import spring.web.java.infrastructure.persistence.PermissionJpaRepository;

@Repository
@RequiredArgsConstructor
public class PermissionRepositoryAdapter implements PermissionRepository {

	private final PermissionJpaRepository jpaRepository;

	@Override
	public Optional<Permission> findById(Long id) {
		return jpaRepository.findById(id);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/adapter/PermissionRepositoryAdapter.kt
package spring.web.kotlin.infrastructure.persistence.adapter

import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Component
import org.springframework.stereotype.Repository
import spring.web.kotlin.domain.repository.PermissionRepository
import spring.web.kotlin.infrastructure.persistence.PermissionJpaRepository

@Component
class PermissionRepositoryAdapter(
    private val jpaRepository: PermissionJpaRepository
) : PermissionRepository {
    override fun findByIdOrNull(id: Long) = jpaRepository.findByIdOrNull(id)
}